### PR TITLE
Feature: custom icon & description for subpacks 

### PR DESCRIPTION
**Pull request by NikSneMC**
Each subpack now can have it's own pack.mcmeta and pack.png

### CODE REVIEW

File: common/src/main/java/io/shardmc/arte/common/pack/meta/file/Namespace.java [hunk 1]
```diff
@@ -5,12 +5,22 @@
...
 
-public record Namespace(String name, Collection<Path> files) implements PackFile {
+public record Namespace(Path root, String name, Collection<Path> files) implements PackFile {
 
     @Override
     public void zip(Zip zip) {
         for (Path file : this.files) {
-            zip.addFile(file);
+            String fileName = file.getFileName().toString();
+            if (fileName.equals("pack.mcmeta") || fileName.equals("pack.png")) {
```
Comment (DrTheodor) [inline common/src/main/java/io/shardmc/arte/common/pack/meta/file/Namespace.java:14]: this if statement looks ugly

File: common/src/main/java/io/shardmc/arte/common/pack/zipper/PackZipper.java [hunk 2]
```diff
@@ -110,7 +110,8 @@ public Collection<BuiltPack> zip(FilterList list, boolean scramble, Consumer<Bui
                             group.zip(zip);
 
                             for (PackFile file : this.defaults) {
-                                file.zip(zip);
+                                if (!group.files().contains(group.getPath().resolve(file.getPath().getFileName())))
```
Comment (DrTheodor) [inline common/src/main/java/io/shardmc/arte/common/pack/zipper/PackZipper.java:113]: wouldnt `file.getPath()` return the same value as `group.getPath().resolve(file.getPath().getFileName())`..?

File: common/src/main/java/io/shardmc/arte/common/zip/Zip.java [hunk 3]
```diff
@@ -24,8 +24,12 @@ public Zip(Path root, Path output, boolean scramble) {
     }
 
     public void addFile(Path path) {
+        this.addFile(path, path);
+    }
+
+    public void addFile(Path path, Path path_zipped) {
```
Comment (DrTheodor) [inline common/src/main/java/io/shardmc/arte/common/zip/Zip.java:30]: no snake case ples

**Review by DrTheodor — CHANGES_REQUESTED** [review]

File: common/src/main/java/io/shardmc/arte/common/pack/zipper/PackZipper.java (hunk 2 above)
Comment (NikSneMC) [reply common/src/main/java/io/shardmc/arte/common/pack/zipper/PackZipper.java:113]: Nope. E.G. `file.getPath()` returns `./arte/resourcepack/pack.mcmeta` and 
`group.getPath().resolve(file.getPath().getFileName())` returns `./arte/resourcepack/assets/tab/pack.mcmeta`

**Pull request merged**